contracts/staking/contracts/src/staking_pools/MixinStakingPool.sol

```diff
@@ -112,6 +118,98 @@ contract MixinStakingPool is
+        IStructs.MakerPoolJoinStatus memory poolJoinStatus = poolJoinedByMakerAddress[makerAddress];
+        if (poolJoinStatus.confirmed) {
+            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
+                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressAlreadyRegistered,
+                makerAddress,
+                getStakingPoolIdOfMaker(makerAddress)
+            ));
+        }
+
+        poolJoinStatus.poolId = poolId;
+        poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
+
+        // Maker has joined to the pool, awaiting operator confirmation
+        emit PendingAddMakerToPool(
+            poolId,
+            makerAddress
+        );
+    }
+
+    /// @dev Adds a maker to a staking pool. Note that this is only callable by the pool operator.
+    /// Note also that the maker must have previously called joinStakingPoolAsMaker.
+    /// @param poolId Unique id of pool.
+    /// @param makerAddress Address of maker.
+    function addMakerToStakingPool(
+        bytes32 poolId,
+        address makerAddress
+    )
+        external
+        onlyStakingPoolOperatorOrMaker(poolId)
+    {
+        _addMakerToStakingPool(poolId, makerAddress);
+    }
+
+    /// @dev Removes a maker from a staking pool. Note that this is only callable by the pool operator or maker.
+    /// Note also that the maker does not have to *agree* to leave the pool; this action is
+    /// at the sole discretion of the pool operator.
+    /// @param poolId Unique id of pool.
+    /// @param makerAddress Address of maker.
+    function removeMakerFromStakingPool(
+        bytes32 poolId,
+        address makerAddress
+    )
+        external
+        onlyStakingPoolOperatorOrMaker(poolId)
+    {
+        bytes32 makerPoolId = getStakingPoolIdOfMaker(makerAddress);
+        if (makerPoolId != poolId) {
+            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
+                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressNotRegistered,
+                makerAddress,
+                makerPoolId
+            ));
+        }
+
+        // remove the pool and confirmation from the maker status
+        IStructs.MakerPoolJoinStatus memory poolJoinStatus = IStructs.MakerPoolJoinStatus({
+            poolId: NIL_POOL_ID,
+            confirmed: false
+        });
+        poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
+        _poolById[poolId].numberOfMakers = uint256(_poolById[poolId].numberOfMakers).safeSub(1).downcastToUint32();
+
+        // Maker has been removed from the pool`
+        emit MakerRemovedFromStakingPool(
+            poolId,
+            makerAddress
+        );
+    }
+
+    /// @dev Returns the pool id of the input maker.
+    /// @param makerAddress Address of maker
+    /// @return Pool id, nil if maker is not yet assigned to a pool.
+    function getStakingPoolIdOfMaker(address makerAddress)
+        public
```

Comment:
Should this just be `internal/private` since `poolJoinedByMakerAddress` is a public variable?

Reply:
We should have public getters for state variables that store a struct in a single word (the autogenerated getters don't play nicely with AbiV2). I'll actually make `poolJoinedByMakerAddress` internal.